server/src/main/java/org/elasticsearch/script/VectorScoreScriptUtils.java

```diff
@@ -52,21 +56,39 @@ public static class ByteDenseVectorFunction extends DenseVectorFunction {
          * @param field The vector field.
          * @param queryVector The query vector.
          */
-        public ByteDenseVectorFunction(ScoreScript scoreScript, DenseVectorDocValuesField field, List<Number> queryVector) {
+        public ByteDenseVectorFunction(
+            ScoreScript scoreScript,
+            DenseVectorDocValuesField field,
+            List<Number> queryVector,
```

Comment:
We should optionally normalize the floating point value. For cosine, we don't want to calculate the magnitude at EVERY comparison. Instead, we should normalize, and then use the `cosineSimilarity(floatValues, false)`

So, the Cosine impl will pass a boolean `true` called `boolean normalizeFloatQuery`, and  all other impls will pass a false.

Reply:
Good point - done